### ISSUE TITLE
Error: did not match C++ signature

### ISSUE THREAD

**Issue by dickreuter** [boostorg/python#279]
I'm calling a pyd as follows:
`equity` = calculator.montecarlo({"3H", "3S"}, {"8S", "4S", "QH", "8C", "4H"}, 2, 10000) `

but it complains that the signature is just a set but not a set of strings:
```

EquityCalculatorMontecarlo.montecarlo(set, set, int, int)
did not match C++ signature:
    montecarlo(class std::set<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::less<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > >, class std::set<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::less<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > >, int, int)

Any suggestions what I can do about it?

**Comment by stefanseefeld** [comment]
You haven't provided any code (such as your EquityCalculatorMontecarlo sources), so I can only guess: It looks like you are wrapping a function taking `std::set<...>` arguments, and you calling it with Python `set` arguments. Have you defined converters for those ? Boost.Python doesn't know the C++ `set` template (much less any particular template instances such as `std::set<std::string>`. You have to define those yourself.

**Comment by dickreuter** [comment]
>> Have you defined converters for those 
correct, I haven't. But what do you mean with converts? Convert from what into what and where?

thanks

**Comment by stefanseefeld** [comment]
You could define something like
```
using string_set = std::set<std::string>;
class_<string_set> sset("string_set");
```
then add a constructor that takes a Python set, as well as a conversion operator that converts to a Python set.

**Comment by dickreuter** [comment]
Ok, I'm a bit new to this so I don't fully understand. What I want is to make this function accessible:
https://github.com/dickreuter/PokerEquityCalculator/blob/master/EquityCalculatorMontecarlo/Scoring.cpp#L240
from python, so it's just a simple function. Is there any way to avoid having to wrap everything into a class with different constructors?

**Comment by stefanseefeld** [comment]
You need to instruct Boost.Python how the function arguments (native Python sets) are to be converted to C++ `std::set<std::string>`. How this is accomplished is explained in the Boost.Python docs at http://boostorg.github.io/python.
